packages/core/src/controllers/subscriber.ts

```diff
@@ -246,9 +270,12 @@ export class Subscriber extends ISubscriber {
     this.pending.delete(params.topic);
   }
 
-  private onResubscribe(id: string, params: SubscriberTypes.Params) {
-    this.addSubscription(id, { ...params, id });
-    this.pending.delete(params.topic);
+  private onBatchSubscribe(subscriptions: SubscriberTypes.Active[]) {
+    if (!subscriptions.length) return;
+    subscriptions.forEach((subscription) => {
+      this.setSubscription(subscription.id, { ...subscription });
```

Comment:
Bit worried about this becoming the new bottleneck but on the I/O-bound level instead of network-bound level. 

Let's see what kind of mileage we get out of this for now and then we can still talk about doing the I/O work (`subscriptions.set`/`topicMap.set` etc) in batch form too if necessary.

Reply:
@bkrem Fair point. Just tested with 3k active subs and performance is degraded _slightly_ on load approx ~1s load time. Also added batching on the batch subscribe for cases where we exceed the 500 max subs payload.